.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs

```diff
@@ -1032,7 +1082,7 @@ private DateTimeResolutionResult ParseQuarter(string text, DateObject referenceD
             }
 
             var beginDate = DateObject.MinValue.SafeCreateFromValue(year, quarterNum * 3 - 2, 1);
-            var endDate = DateObject.MinValue.SafeCreateFromValue(year, quarterNum * 3 + 1, 1);
+            var endDate = DateObject.MinValue.SafeCreateFromValue(year, quarterNum * 3, 1).AddMonths(1);
```

Comment:
It would be better to change all cases to use such methods instead of manually adding/subtracting months.
Also, please declare constants to represent 3 and 6 in such equations. 3 can be TrimesterMonthCount and 6 SemesterMonthCount or similar.
You can also simplify the calculations and place inside a helper method that safely computes start/end of the period you need.